### PR TITLE
Migrate Scenario vector and index map to a single UUID:Scenario map

### PR DESCRIPTION
**Pull request by greenscientist**
Not too many changes at this point for that one

### CODE REVIEW

File: connection_scan_algorithm/src/route_parameters.cpp [hunk 1]
```diff
@@ -271,7 +272,7 @@ namespace TrRouting
 
     return RouteParameters(std::make_unique<TrRouting::Point>(origin->latitude, origin->longitude),
       std::make_unique<TrRouting::Point>(destination->latitude, destination->longitude),
-      *scenario,
+                           scenario.value().get(),
```
Comment (tahini) [inline connection_scan_algorithm/src/route_parameters.cpp]: alignment here

File: tests/connection_scan_algorithm/route_parameters_test.cpp [hunk 2]
```diff
@@ -28,28 +27,21 @@ class RouteParametersFixtureTests : public BaseParametersFixtureTests
 
         // Create a valid scenario, services and other data don't have to exist, we are creating parameters, not using them
         std::unique_ptr<TrRouting::Scenario> scenario = std::make_unique<TrRouting::Scenario>();
```
Comment (tahini) [inline tests/connection_scan_algorithm/route_parameters_test.cpp]: Do you still need this variable here?

**Review by tahini — APPROVED** [review]
That's much easier to review than the node one! :D

File: tests/connection_scan_algorithm/route_parameters_test.cpp (hunk 2 above)
Comment (greenscientist) [reply tests/connection_scan_algorithm/route_parameters_test.cpp]: nope, good point. (Soon, the clean up of -Wall will allow us to see this

**Pull request merged**